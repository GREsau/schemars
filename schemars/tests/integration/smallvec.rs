use crate::prelude::*;
use smallvec1::{smallvec, SmallVec};

#[test]
fn smallvec() {
    test!(SmallVec<[usize; 2]>)
        .assert_identical::<Vec<usize>>()
        .assert_allows_ser_roundtrip([smallvec![], smallvec![1, 2, 3, 4, 5]])
        .assert_matches_de_roundtrip(arbitrary_values());
}
