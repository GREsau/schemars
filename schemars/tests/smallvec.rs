mod util;
use smallvec1::SmallVec;
use util::*;

#[test]
fn smallvec() -> TestResult {
    test_default_generated_schema::<SmallVec<[String; 2]>>("smallvec")
}
