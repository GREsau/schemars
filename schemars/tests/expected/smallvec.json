{
  "$schema": "https://json-schema.org/draft/2020-12/schema",
  "title": "Array_of_string",
  "type": "array",
  "items": {
    "type": "string"
  }
}